Oracle/PCORNetLoader_ora.sql

```diff
@@ -1945,56 +1084,142 @@ insert into i2pReport values(v_runid, SYSDATE, 'Dispensing',	null,		dispensings,
+-- procedures above.
+--------------------------------------------------------------------------------
+create or replace procedure PCORNetPostProc as
 begin
----pcornetclear;
-PCORNetDemographic;
-PCORNetEncounter;
-PCORNetDiagnosis;
-PCORNetCondition;
-PCORNetProcedure; 
-PCORNetVital;
-PCORNetEnroll;
-PCORNetLabResultCM;
-PCORNetPrescribing;
-
-/* ORA-04068: existing state of packages has been discarded
-ORA-04065: not executed, altered or dropped stored procedure "PCORNETDISPENSING"
-ORA-06508: PL/SQL: could not find program unit being called: "PCORNETDISPENSING"
-ORA-06512: at "PCORNETLOADER", line 14
-ORA-06512: at line 2
-04068. 00000 -  "existing state of packages%s%s%s has been discarded"
-*Cause:    One of errors 4060 - 4067 when attempt to execute a stored
-           procedure.
-*Action:   Try again after proper re-initialization of any application's
-           state.
-
-The above error only happens when we call PCORNetDispensing _and_ PCORNetPrescribing
-from within pcornetloader.  When running either individually, the error does not
-happen.
-
-Skipping dispensing as per gpc-dev notes:
-http://listserv.kumc.edu/pipermail/gpc-dev/attachments/20160223/8d79fa70/attachment-0001.pdf
-> LV: the dispensing side [?] is not mandatory? we just did Rx, since that
-> what we have in our i2b2
-*/
-PCORNetDispensing;
-PCORNetDeath;
-PCORNetHarvest;
-
-end pcornetloader;
+  
+  /* Copy providerid from encounter table to diagnosis, procedures tables.
+  CDM specification says:
+    "Please note: This is a field replicated from the ENCOUNTER table."
+  */
+  merge into diagnosis d
+  using encounter e
+     on (d.encounterid = e.encounterid)
+  when matched then update set d.providerid = e.providerid;
+  
+  merge into procedures p
+  using encounter e
+     on (p.encounterid = e.encounterid)
+  when matched then update set p.providerid = e.providerid;
+  
+  merge into prescribing p
+  using encounter e
+     on (p.encounterid = e.encounterid)
+  when matched then update set p.rx_providerid = e.providerid;
+  
+  /* Currently in HERON, we have hight in cm and weight in oz (from visit vitals).
+  The CDM wants height in inches and weight in pounds. */
+  update vital v set v.ht = v.ht / 2.54;
+  update vital v set v.wt = v.wt / 16;
+  
+  /* Remove rows from the PRESCRIBING table where RX_* fields are null
+     TODO: Remove this when fixed in HERON
+   */
+  delete
+  from prescribing
+  where rx_basis is null
+    and rx_quantity is null
+    and rx_frequency is null
+    and rx_refills is null
+  ;
+  
+end PCORNetPostProc;
 /
 
 
-BEGIN
-pcornetloader; --- you may want to run sql statements one by one in the pcornetloader procedure :)
-END;
+create or replace PROCEDURE PCORNetLoader(start_with VARCHAR2) AS
+begin
+  if start_with in ('PCORNetDemographic') then
+    PCORNetDemographic;
+  end if;
+  
+  if start_with in ('PCORNetDemographic', 'PCORNetEncounter') then
+    PCORNetEncounter;
+  end if;
+  
+  if start_with in (

```

Comment:
Yeah, this guy is magic string city.  Your suggestion could clean this procedure up a bit w/ the right documentation. I'm a little hesitant to do so because the name wouldn't get logged in the Jenkins console, but I know if that's a strong reason not to.